internal/testutil/usersync.go

```diff
@@ -0,0 +1,44 @@
+package testutil
```

Comment:
I don't like the idea of a "testutil" package...however, it's in the internal directory which is not exposed to the public api.

Thoughts?

Reply:
I'm a fan of code sharing. In the vast majority of cases, utils are much better than duplicate code.

I'm not too worried about the package API, either, because this is an application  and not a library. Our REST APIs are _very_ important... but the package APIs are less so because we don't expect people to use this code by importing it. Any package structure within this project is purely for our own sanity.

That said: if you take a step back, do you think these utilities really improve things? Compare these code snippets (loosely based on the `sovrn` syncer).

Without utils (using [testify](https://godoc.org/github.com/stretchr/testify/assert)):
```
info := syncer.GetUsersyncInfo("0", "")
assert.Equal(t, "//ap.lijit.com/pixel?redir=external.com%2Fsetuid%3Fbidder%3Dsovrn%26gdpr%3D0%26gdpr_consent%3D%26uid%3D%24UID", info.URL)
assert.Equal(t, "redirect", info.Type)
assert.False(t, info.SupportCORS)
assert.Equal(t, 13, syncer.GDPRVendorID())
```

With utils:
```
u := testutil.UsersyncTest(t, syncer, syncer.GetUsersyncInfo("0", ""))
u.Assert(
  "//ap.lijit.com/pixel?redir=external.com%2Fsetuid%3Fbidder%3Dsovrn%26gdpr%3D0%26gdpr_consent%3D%26uid%3D%24UID",
  "redirect",
  13,
  false,
)
```

Advantages of no utils:
- Fewer lines of code
- Clearer to read
- No need to implement the utils function
- No need to find a good package location for the utils implementation

Advantages of utils:
- When a field gets added to the usersyncer, we'd update `u.Assert` and nobody's tests would compile until _every_ adapter is running the assertion

Reply:
I was mostly avoiding adding another dependency, didn't realize testify was already being used so I will update it.

One quick comment about your statement: 
> I'm not too worried about the package API, either, because this is an application and not a library.
Treating the components as libraries and stabilizing the API will allow the community to use the code here and contribute back improvements. Keeping it isolated as an application will cause users to fork and less likely to contribute anything back (I know I sound like a broken record at this point, however I really feel like this will improve the health of the project).

Reply:
That's an interesting perspective... I wouldn't have thought this would make it hard for people to contribute to the project.

You're not a broken record at all. From your previous comments, I thought your main issue was that you or someone you knew was trying to add a Bidder and finding it annoying to change lots of different places in the code. I didn't realize you planned to import some packages and rely on them directly. Thanks for the perspective.

I do agree that the healthier an application architecture is, the more it resembles a bunch of libraries with a thin layer tying them all together... so whether or not we ultimately fold package APIs into our "v1" compatibility, it's definitely helpful to organize the code as if we plan to.